hybridcloud/backends/azure_postgresqlflexible.py

```diff
@@ -88,8 +87,10 @@ def create_or_update_server(self, namespace, name, spec, password, admin_passwor
         if public_access:
             network = None
```

Comment:
Have you checked what happens when this field gets changed in the manifests? Are the resources created in the else branch properly cleaned up?